cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy

```diff
@@ -0,0 +1,157 @@
+import groovy.transform.Field
+import org.artifactory.fs.ItemInfo
+import org.artifactory.repo.RepoPath
+import static com.google.common.collect.Multimaps.forMap
+// jobs to trigger based on cron or interval
+jobs {
+    // Every Min Run cron Regex  depreciation run
+    dailyDataDepreciation(cron: "0 0/1 * 1/1 * ? *") {
+        depreciateData("repositoryname-local")
+    }
+}
+def depreciateData(repoName){
+    
+    log.info "Starting data depreciation job for repository " + repoName // This will log in the LogFile
+    
+    // search given repo name for folders with property DataDeprecation = true 
+    def repoRootPath = RepoPathFactory.create(repoName, "/")
+    List<RepoPath> deprecatePaths = searches.itemsByProperties(forMap(['DataDeprecation': 'true']))
+    // call the corresponding method to handle deprecation
+    for(RepoPath repoPath : deprecatePaths){
+        log.info "Depreciating data for path " + repoPath
+        
+
+        String deprecateMethodStr = repositories.getProperty(repoPath, 'DataDeprecationMethod')   //version:4 This is the Property on the artifact
+if (deprecateMethodStr !=null)
+{
+    String[] depreciateMethodOpts = deprecateMethodStr.split(':')   //  (version) : (4) 
+
+    String[] depreciateMethodArgs = depreciateMethodOpts[1]
+moduleMajorVersion(repoPath,repoName,depreciateMethodArgs)
+  
+}else{
+    println "No Such Property Defined on the folder"
+}
+      
+    }
+}
+/**
+    Deprecate files by Module name and major version, removing by build number
+    Keep last x number of semantic versions of an Artifact
+    args is assumed to be a single integer value (but as a string) for number of files to keep
+*/
+def moduleMajorVersion(RepoPath repoPath, repoName,String[] args){
+def int numToKeep =0
+args.each{
+    log.info "Depreciating data using method moduleMajorVersion and args " + it
+    println "Depreciating data using method moduleMajorVersion and args " + it
+    numToKeep=it
+}
+    println "Starting data depreciation job for repository " + repoName
+    log.info "Starting data depreciation job for repository " + repoName
+    // search given repo name for folders with property DataDeprecation = true
+    def repoRootPath = RepoPathFactory.create(repoName, "/")
+    List<RepoPath> deprecatePaths = searches.itemsByProperties(forMap(['DataDeprecation': 'true']))
+    List<ItemInfo> folderParent 
+    List<String> fillVersions =[""]
+    def int numArtifacts
+    // call the corresponding method to handle deprecation
+    for(RepoPath rPath : deprecatePaths){
+        println "Depreciating data for path " + rPath
+        log.info "Depreciating data for path " + rPath
+         def parent = rPath.parent
+       println "Parent Path " + parent
+
+       folderParent= repositories.getChildren(rPath)
+      for (child in folderParent) {
+         RepoPath childRepoPath = child.getRepoPath()
+        def artifactVersion = repositories.getProperty(childRepoPath,"build.number")
```

Comment:
You are getting the artifact version from the `build.number` property and assuming that it has the `x.x.x.x` format. What happens if that is not true?